Auth/Auth.Infrastructure/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -26,11 +27,8 @@ private static void AddDbContext(this IServiceCollection services, IConfiguratio
 
         private static void AddRepositories(this IServiceCollection services)
         {
-            services.AddScoped<IUserRepository, UserRepository>();
-            services.AddScoped<IUserRoleRepository, UserRoleRepository>();
-            services.AddScoped<IAccountDataRepository, AccountDataRepository>();
-            services.AddScoped<IPublisherRepository, PublisherRepository>(); 
-            services.AddScoped<IAuthorRepository, AuthorRepository>();
+            services.AddScoped<IRepository<User>, Repository<User>>();
+            services.AddScoped<IRepository<UserRole>, Repository<UserRole>>();
```

Comment:
can be done like `services.AddScoped(typeof(IRepository<>), typeof(Repository<>));`